test/disk_uploader_test.go

```diff
@@ -0,0 +1,112 @@
+	kubeclientcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+	kubevirtcliv1 "kubevirt.io/client-go/kubecli"
+	cdiv1beta1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/tests/libdv"
+
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/constants"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/framework"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/runner"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/testconfigs"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/vm"
+)
+
+var _ = Describe("Run disk-uploader", func() {
+	f := framework.NewFramework()
+
+	BeforeEach(func() {
+		if f.TestOptions.SkipDiskUploaderTests {
+			Skip("skipDiskUploaderTests is set to true, skipping tests")
+		}
+	})
+
+	It("Extracts disk from VM and upload to container registry", func() {
+		imageDestination := os.Getenv("REGISTRY_IMAGE_DESTINATION")
+		Expect(imageDestination).ToNot(BeEmpty())
+
+		registryCredentials, err := createRegistryCredentials(f.CoreV1Client, f.DeployNamespace)
+		Expect(err).ToNot(HaveOccurred())
+
+		cirrosVm, err := createCirrosVM(f.KubevirtClient, f.DeployNamespace)
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = vm.WaitForVM(f.KubevirtClient, f.DeployNamespace, cirrosVm.Name, "", constants.Timeouts.WaitForVMStart.Duration, false)
+		Expect(err).ToNot(HaveOccurred())
+
+		config := &testconfigs.DiskUploaderTestConfig{
+			TaskRunTestConfig: testconfigs.TaskRunTestConfig{},
+			TaskData: testconfigs.DiskUploaderTaskData{
+				ExportSourceKind: "vm",
+				ExportSourceName: cirrosVm.Name,
+				VolumeName:       cirrosVm.Spec.Template.Spec.Volumes[0].DataVolume.Name,
+				ImageDestination: imageDestination,
+				SecretName:       registryCredentials.Name,
+			},
+		}
+		f.TestSetup(config)
+
+		runner.NewTaskRunRunner(f, config.GetTaskRun()).
+			CreateTaskRun().
+			ExpectSuccess().
+			ExpectLogs(config.GetAllExpectedLogs()...).
+			ExpectResults(nil)
+	})
+})
+
+func createRegistryCredentials(client kubeclientcorev1.CoreV1Interface, namespace string) (*corev1.Secret, error) {
+	registryKeyId := os.Getenv("REGISTRY_ACCESS_KEY_ID")
+	Expect(registryKeyId).ToNot(BeEmpty())
+
+	registryKey := os.Getenv("REGISTRY_SECRET_KEY")
+	Expect(registryKey).ToNot(BeEmpty())
+
+	v1Secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: constants.E2ETestsRandomName("disk-uploader-credentials"),
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"accessKeyId": []byte(registryKeyId),
+			"secretKey":   []byte(registryKey),
+		},
+	}
+
+	return client.Secrets(namespace).Create(context.Background(), v1Secret, metav1.CreateOptions{})
+}
+
+func createCirrosVM(client kubevirtcliv1.KubevirtClient, namespace string) (*v1.VirtualMachine, error) {
+	dataVolume := libdv.NewDataVolume(
+		libdv.WithRegistryURLSource("docker://quay.io/kubevirt/cirros-container-disk-demo"),
```

Comment:
Please do not use `libdv`. It lives under `tests/libdv` which might be subject to change. Only `libvmi` can be considered stable.

Reply:
@0xFelix Don't you think that this is an issue that such important libs for external projects tests living under `kubevirt/kubevirt`, especially under `tests/libdv`?

Reply:
@0xFelix @ksimon1 I raised this discussion in the code quality meeting. I got an approval from @akalenyu to relocate `libdv` from `tests` to `pkg` - needs to see how it'll work.

Reply:
Done.

Reply:
This [revision](https://github.com/kubevirt/kubevirt-tekton-tasks/compare/ef903df540d552fc6056d927f6fd74fb39e95f29..8350664b97acdf4964327a2477387254d29d611a) removes the usage of `tests/libdv` from `kubevirt/kubevirt` repository.